stripe/checkoutCompleted.js

```diff
@@ -0,0 +1,59 @@
+import Stripe from 'stripe';
+import { v1 as uuidv1 } from 'uuid';
+import { badRequest, failure, success } from '../libs/response-lib';
+import * as dynamoDbLib from '../libs/dynamodb-lib';
+
+export const main = async event => {
+  const stripeSignature = event.headers['Stripe-Signature'];
+  const stripe = Stripe(process.env.STRIPE_API_SECRET_KEY);
+
+  let webhookEvent;
+
+  console.log('Verify the event with Stripe');
+  try {
+    webhookEvent = stripe.webhooks.constructEvent(
+      event.body,
+      stripeSignature,
+      process.env.STRIPE_WEBHOOK_CHECKOUT_COMPLETED_SECRET_KEY // We are getting this one from Stripe Dashboard, when a webhook is created
+    );
+  } catch (err) {
+    console.error(err);
+    return badRequest(`Webhook Error: ${err.message}`);
```

Comment:
Do we know if request with all parameters would be logged? I'm trying to guess how to be fault-tolerant. To be able to restore the coupon even if our source code failed and a coupon is not saved to the DB.

Reply:
We can do `console.error(event.body)` the body contains the info regarding the purchase

Reply:
I suppose `console.log` would be great 🙃

Reply:
Would it be useful to have request headers in the log as well? I don't know if stripe sends something useful in headers. it could help to differentiate lawful requests from stripe from hackers' requests

Reply:
btw, I meant to log the data for all requests, not only one with those we have errors
I could imaging AWS could have some similar log built-in 🤷‍♂

Reply:
If I understand you correctly, we should log everything coming from Stripe, right? Body and Headers included?